opensearch-operator/pkg/builders/cluster.go

```diff
@@ -65,6 +65,19 @@ func NewSTSForNodePool(
 		}
 	}
 
+	for _, role := range selectedRoles {
+		found := false
+		for _, availRole := range availableRoles {
+			if role == availRole {
+				found = true
+				break
+			}
+		}
+		if !found {
+			fmt.Printf("NOTICE! -- %s is not an available role\n", role)
```

Comment:
Do not directly print something. Always use a logger.

Reply:
Hey @swoehrl-mw 
just added the logger but somehow it failed the test, do you have a clue why it affect it ?

Reply:
@idanl21 The logger instance you are provoding in the tests is not correctly initialized.
You can use this function to create a logger:
```go
func dummyLogger() logr.Logger {
	var dlog = log.NewDelegatingLogSink(log.NullLogSink{})
	return logr.New(dlog)
}
```

and then in your tests replace `logr.Logger{}` with `dummyLogger()` and it should be fine.